### PR TITLE
Tests

### LINKED ISSUE #58: [Bug]: Language fallback no longer working
### What did you expect? 🧐

An undefined language should result in the announcement using the provided fallback.

### What actually happened? 😵‍💫

No announcement is made if the language is not defined in the options. This started happening in 4.5.1, specifically after [this pull request](https://github.com/swup/a11y-plugin/pull/46) was merged. The part that seemed redundant was actually the relevant implementation of the fallback: `announcements['*']`.

I'll make a note to fix and test this in the test branch that's currently in progress, just flagging it here so we don't forget.

### Swup and plugin versions 📝

- swup: *
- a11y plugin: 4.5.1

### What browsers are you seeing the problem on? 🧭

_No response_

### Relevant log output 🤓

_No response_

### URL to minimal reproduction 🔗

https://github.com/swup/a11y-plugin/pull/46/files

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for related problems
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

### PR DESCRIPTION
**Pull request by daun**
**Description**

- Add tests and fix a few issues discovered during testing
- Turns out we can test the whole plugin just in vite, same as the head plugin
- Avoiding Playwright for as long as possible makes things much simpler and faster 🧃

**Changes**

- Straighten out a few types
- Make `swup.announce()` return a Promise to simplify testing
- Check reduced-motion option just-in-time to simplify testing
- Extract announcement delay into documented variable

**Fixes**

- Re-implement language fallback
- Fix selector for `[inert]` autofocus element

**Checks**

- [x] The PR is submitted to the `master` branch
- [x] The code was linted before pushing (`npm run lint`)
- [x] All tests are passing (`npm run test`)
- [x] New or updated tests are included
- [ ] ~~The documentation was updated as required~~

**Notes**

Closes #58

### CODE REVIEW

File: tests/unit/announcements.test.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			const announcer = new Announcer();
+			await announcer.announce('Hello');
+			expect(announcer.region.textContent).toBe('Hello');
+		});
+
+		it('modifies the announcement to allow duplicate messages', async () => {
+			const announcer = new Announcer();
+			const region = announcer.region;
+
+			await announcer.announce('Hello');
```
Comment (hirasso) [inline tests/unit/announcements.test.ts:56]: That test description is a bit confusing to me. I suspect what it actually tests is that an unchanged announcement won't land as `${announcement}.` in the region? And that was a "hack" to make sure the region actually dispatches a change?

**Review by hirasso — APPROVED** [review]
Amazing work! It's beautiful reading through the tests.

File: tests/unit/announcements.test.ts (hunk 1 above)
Comment (daun) [reply tests/unit/announcements.test.ts:56]: I think "slightly modifies the announcement to allow announcing the same message twice in a row" would be the appropriate description 🤠 It's checking that it actually **will** land as `{message}.` in the region, which is the hack required to circumvent screen readers from ignoring technically unchanged live regions. Is the test description still confusing then?

Comment (daun) [reply tests/unit/announcements.test.ts:56]: Or maybe "modifies each announcement to make it unique"?

File: src/announcements.ts [hunk 2]
```diff
@@ -15,37 +15,44 @@ export class Announcer {
...
-				message = `${message}.`;
-			}
-			// Clear before announcing
-			this.region.textContent = '';
-			this.region.textContent = message;
-		}, delay);
+	announce(message: string, delay: number = 0): Promise<void> {
+		return new Promise((resolve) => {
+			setTimeout(() => {
+				// // Fix screen readers not announcing the same message twice
```
Comment (hirasso) [inline src/announcements.ts:29]: Maybe this code comment could be optimized (apart from the double `//`):

```js
// Fix screen readers not announcing the same message again
```

File: tests/unit/announcements.test.ts (hunk 1 above)
Comment (hirasso) [reply tests/unit/announcements.test.ts:56]: Aaah, now I understand! The announcer would switch back and forth between the message with and without the dot? That's amazingly clever (and also hard to understand in 3 months) 😃 

...I like both your suggestions for the test description better then the original. With a slight preference for the shorter one. Happy to let you choose :)

File: src/announcements.ts (hunk 2 above)
Comment (daun) [reply src/announcements.ts]: True, updated it with a description of the actual intent.

File: tests/unit/announcements.test.ts (hunk 1 above)
Comment (daun) [reply tests/unit/announcements.test.ts:56]: Correct! It toggles the dot to allow that. I've used the shorter description for the test :)

**Review by hirasso — APPROVED** [review]
🍰

**Pull request merged**